### PR TITLE
Improved contributing_to_vhive.md: More detailed rationale and technique for Rebase & Merge

### PR DESCRIPTION
**Pull request by googlercolin**
## Summary

Improved the descriptions in the "Rebasing on the Main Branch" and "Maintaining a Clean Main Branch & Avoiding Merge Commits" sections.

## Implementation Notes :hammer_and_pick:

* Provided an example of the consequence of a messy merge without rebasing
* Included `git checkout your-branch` in the instructions on how to rebase on main
* Included clearer motivations behind maintaining a clean main branch and avoiding merge commits

## External Dependencies :four_leaf_clover:

* N/A

## Breaking API Changes :warning:

* N/A

*Simply specify none (N/A) if not applicable.*

### CODE REVIEW

**Review by yankai14 — DISMISSED** [review]

**Comment by leokondrashov** [comment]
@googlercolin Please, sign your commit and update the branch.

**Review by ustiugov — CHANGES_REQUESTED** [review]

**Comment by ustiugov** [comment]
Please sign the commit. 

BTW, do we mention the requirement to sign commits in the doc?

File: docs/contributing_to_vhive.md [hunk 1]
```diff
@@ -1,107 +1,157 @@
...
 ```
-              A'--B'--C' branch
+              A'--B'--C' feature
              /
 D---E---F---G master
 ```
-For more detail check out [the git rebase documentation](https://git-scm.com/docs/git-rebase).
 
-To rebase on main simply checkout your branch and use the following:
+As you can see, your branch's commits (`A`, `B`, `C`) now build forward from the latest `main` commit (`G`). This linearizes the history and eliminates the need for merge commits.
```
[the remark this answers is not in the document]
Comment (googlercolin) [reply docs/contributing_to_vhive.md:98]: Do you mean break line 98 into two lines? Would the same apply for other lines with several sentences? Or does this comment pertain only to this section?

**Comment by googlercolin** [comment]
@ustiugov, I have signed the commit and pushed it here. The spellcheck linter is a bit weird because "linearizes" is an actual word in the Oxford dictionary. "Upstream's" was also flagged out. I've changed them nonetheless.

For the part on signing commits, I think I will make another PR to include it in the doc as it is not there at the moment.

**Review by ustiugov — CHANGES_REQUESTED** [review]
some lines are still too long

**Review by ustiugov — APPROVED** [review]

**Pull request merged**